cfme/tests/infrastructure/test_host.py

```diff
@@ -529,6 +536,44 @@ def test_add_ipmi_refresh(appliance, setup_provider):
     assert view.ipmi_address.read() == ipmi_address
 
 
+@pytest.fixture
+def host(appliance, provider):
+    host_collection = provider.hosts
+    return random.choice(host_collection.all())
+
+
+HOST_DETAIL_ROWS = [
+    ("VMM Information", HostVmmInfoView),
+    ("Operating System", HostOsView),
+    ("Devices", HostDevicesView),
+    ("Networks", HostNetworkDetailsView),
+    ("Storage Adapters", HostStorageAdaptersView),
+    ("Services", HostServicesView),
+    ("Print or export", HostPrintView),
+]
+
+
+@test_requirements.infra_hosts
+@pytest.mark.parametrize("row,view", HOST_DETAIL_ROWS,
+    ids=[rel[0] for rel in HOST_DETAIL_ROWS])
+def test_infrastructure_hosts_viewing(appliance, setup_provider, host, row, view):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: high
+        initialEstimate: 1/6h
+    """
+    # Except for OS, Network, and print we need to check if we have any of the row items.
+    if row not in ["Operating System", "Networks", "Print or export"]:
+        view = navigate_to(host, "Details")
+        pytest.skip(f"No row item present for {row}")
+    view = navigate_to(host, row)
+    if row == "Print or export":
+        handle_extra_tabs(view)
+    assert view.is_displayed
```

Comment:
This got a bit screwed up by the suggestion merged as 333b9645241fa589c5b32e114fd7d79b45e0af47

```suggestion
    @request.addfinalizer
    def _finalizer():
        if row == "Print or export":
            handle_extra_tabs(view)
    
    try:
        if row not in ["Operating System", "Networks", "Print or export"]:
            view = navigate_to(host, "Details")
            if view.entities.summary(table).get_text_of(row) == '0':
                pytest.skip(f"No items present for host relationship {row}")
        view = navigate_to(host, row)
    except TimedOutError:
        pytest.fail('Timed out navigating to host relationship {row}')    

```